### PR TITLE
Star 229 list box mdx

### PR DESCRIPTION
**Pull request by KaanDarcey**
### Purpose 🚀
To create an mdx document that demonstrates to users how to use the ListBox component.

### Notes ✏️
- Created ListBox.mdx
- Created ListBox.stories.mdx
- Added requirement to .storybook/config.js

### Updates 📦
- [ ] MAJOR (breaking) change to _these packages_
- [x] MINOR (backward compatible) change to _these packages_
- [ ] PATCH (bug fix) change to _these packages_

### Storybook 📕
http://storybooks.highbond-s3.com/paprika/STAR-229-ListBox-mdx/?path=/story/listbox-docs--page

### Screenshots 📸
N/A

### References 🔗
N/A

### CODE REVIEW

File: packages/ListBox/stories/ListBox.mdx [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+
+## Usage
+
+### Basic Single ListBox
+
+The single ListBox creates a basic ListBox in which a user can make one selection at a time.
+
+```js
+    <ListBox></ListBox>
+```
```
Comment (oscarkwan) [inline packages/ListBox/stories/ListBox.mdx:43]: This part is probably not necessary.

File: packages/ListBox/stories/ListBox.mdx [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+
+To add new list options simply add the child element ``` <ListBox.option></ListBox.option>``` to your ListBox.
+
+```js
+    <ListBox>
+      <ListBox.Option>Option 1</ListBox.Option>
+      <ListBox.Option>Option 2</ListBox.Option>
+    </ListBox>
```
Comment (oscarkwan) [inline packages/ListBox/stories/ListBox.mdx:51]: Can we make sure all the code examples have the same indentation? I see some inconsistencies with the code blocks

File: packages/ListBox/stories/ListBox.mdx [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+
+### Basic Single ListBox
+
+The single ListBox creates a basic ListBox in which a user can make one selection at a time.
+
+```js
+    <ListBox></ListBox>
+```
+
+To add new list options simply add the child element ``` <ListBox.option></ListBox.option>``` to your ListBox.
```
Comment (oscarkwan) [inline packages/ListBox/stories/ListBox.mdx:45]: <ListBox.Option>

File: packages/ListBox/stories/ListBox.mdx [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+      <ListBox.Option>Option 2</ListBox.Option>
+    </ListBox>
+```
+#### Result
+
+<ListBox><ListBox.Option>Option 1</ListBox.Option><ListBox.Option>Option 2</ListBox.Option></ListBox>
+
+### Adding a Filter to the ListBox
+
+You can filter your ListBox by adding the ```ListBox.Filter```. This adds an input in which users can type to filter through the list options.
```
Comment (oscarkwan) [inline packages/ListBox/stories/ListBox.mdx:59]:
```suggestion
You can filter your ListBox by adding the ```<ListBox.Filter />```. This adds an input in which users can type to filter through the list options.
```

File: packages/ListBox/stories/ListBox.mdx [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+    </ListBox>
+```
+
+#### Result
+
+<ListBox><ListBox.Filter /><ListBox.Option>Option 1</ListBox.Option><ListBox.Option>Option 2</ListBox.Option></ListBox>
+
+### Adding a Footer to the ListBox
+
+You can add a footer to your ListBox by adding the ```ListBox.Footer```. This adds a footer to the bottom of the listBox when it is expanded. Users can select to Accept, Cancel, or Clear their current list option selections.
```
Comment (oscarkwan) [inline packages/ListBox/stories/ListBox.mdx:75]:
```suggestion
You can add a footer to your ListBox by adding the ```<ListBox.Footer />```. This adds a footer to the bottom of the ListBox when it is expanded. Users can select to Accept, Cancel, or Clear their current list option selections.
```

File: packages/ListBox/stories/ListBox.mdx [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+import { images } from "./fixtures/images";
+import * as characters from "./fixtures/characters";
+
+import * as Multi from "./examples/multi";
+import * as Single from "./examples/single";
+import FilterSelect from "./moreExamples/filterSelect/FilterSelect";
+
+# ListBox
+
+The ListBox component allows users to select one or more options from a list contained within the component.
```
[the remark this answers is not in the document]
Comment (KaanDarcey) [reply packages/ListBox/stories/ListBox.mdx:14]: Now that you mention it, that information does seem somewhat redundant. I'll remove it.

File: packages/ListBox/stories/ListBox.mdx [hunk 7]
```diff
@@ -0,0 +1,284 @@
...
+  </ListBox>
+```
+
+#### Result
+
+<ListBox isInline onChange={activeOptionIndex => {console.log(activeOptionIndex);}}>{characters.antiHeroesRaw.map((item, index) => {return (<ListBox.Option key={item.label} defaultIsSelected={index === 4}>{item.label}</ListBox.Option>);})}</ListBox>
+
+### Multi ListBox
+
+The `<ListBox>` component provides a prop `isMulti`. It allows users to select multiple items from a drop-down list.
```
Comment (oscarkwan) [inline packages/ListBox/stories/ListBox.mdx:178]: perhaps stay away from "drop-down" So it doesn't get confused with our DropdownMenu component.

**Comment by oscarkwan** [comment]
Can we have an example / tutorial talking a bit about the `<ListBox.Trigger>`

**Comment by nahumzs** [comment]
do we need anything else on this branch or can we 👍?

**Pull request merged**